### PR TITLE
Test mvnw

### PR DESCRIPTION
**Pull request by mikebell90**

### CODE REVIEW

File: .mvn/wrapper/maven-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.5.0/apache-maven-3.5.0-bin.zip
```
Comment (pennello) [inline .mvn/wrapper/maven-wrapper.properties:1]: Yikes, it's unfortunate that we'd have to duplicate the version of Maven across each of our repos.  Is it possible to pull this from elsewhere, like the parent POM?

Comment (pennello) [inline .mvn/wrapper/maven-wrapper.properties:1]: Nit: missing terminating newline.

File: mvnw [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    MAVEN_PROJECTBASEDIR=`cygpath --path --windows "$MAVEN_PROJECTBASEDIR"`
+fi
+
+WRAPPER_LAUNCHER=org.apache.maven.wrapper.MavenWrapperMain
+
+exec "$JAVACMD" \
+  $MAVEN_OPTS \
+  -classpath "$MAVEN_PROJECTBASEDIR/.mvn/wrapper/maven-wrapper.jar" \
+  "-Dmaven.home=${M2_HOME}" "-Dmaven.multiModuleProjectDirectory=${MAVEN_PROJECTBASEDIR}" \
+  ${WRAPPER_LAUNCHER} $MAVEN_CONFIG "$@"
```
Comment (pennello) [inline mvnw:225]: Oh wow, so we'd have to include this in every repo?

File: mvnw.cmd [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+if exist "%HOME%\mavenrc_post.bat" call "%HOME%\mavenrc_post.bat"
+if exist "%HOME%\mavenrc_post.cmd" call "%HOME%\mavenrc_post.cmd"
+:skipRcPost
+
+@REM pause the script if MAVEN_BATCH_PAUSE is set to 'on'
+if "%MAVEN_BATCH_PAUSE%" == "on" pause
+
+if "%MAVEN_TERMINATE_CMD%" == "on" exit %ERROR_CODE%
+
+exit /B %ERROR_CODE%
```
Comment (pennello) [inline mvnw.cmd:143]: I think we could safely omit the Windows compatibility.

File: .idea/compiler.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      </profile>
+    </annotationProcessing>
+    <bytecodeTargetLevel>
+      <module name="otj-pausedetector" target="1.8" />
+    </bytecodeTargetLevel>
+  </component>
+  <component name="JavacSettings">
+    <option name="ADDITIONAL_OPTIONS_STRING" value="-parameters" />
+  </component>
+</project>
```
Comment (pennello) [inline .idea/compiler.xml:19]: Please omit your IDE files; rebase/force/whatever to get rid of this.

**Review by pennello — COMMENTED** [review]
Interesting, although I'm concerned about the per-repo additions we'd have to duplicate.

**Comment by pennello** [comment]
Another thought is that it would be unfortunate to have to depend on downloading a remote file (the linked Maven zip).  (Although, yeah, I know we already depend on the other remote dependencies.  Those are at least, in theory, locally-handle-able by having a mirror of Maven central.)

**Comment by stevenschlansker** [comment]
Is the point of this just to side-step the need to configure Maven on each developer's workstation?  I'm not sure this is simpler than just configuring Maven right.  But maybe I missed the point?

**Comment by mikebell90** [comment]
Closing this since no one including me is excited about the option. We can always dig up the PR if needed later.

**Pull request closed without merging**